### ISSUE TITLE
Shebang in post_install scripts not recognized

### ISSUE THREAD

**Issue by guicho271828** [awslabs/cfncluster#103]
Despite as promised to be in http://cfncluster.readthedocs.io/en/latest/pre_post_install.html

```
The code will be executed as root and can be in any script language
 supppoted by the cluster OS, typically bash or python.
```

**Comment by guicho271828** [comment]
It has a executable flags on, to be sure.

**Comment by dougalb** [comment]
Hi, I have tested using your script as `post_install` and don't see any issues. I would suggest using the `--no-rollback` option with `create` and then login to the **MasterServer** and post the _/var/log/cfn-init.log_ here.

**Comment by guicho271828** [comment]
thank you, I'll soon try `--norollback`.

**Comment by guicho271828** [comment]
Please forgot the above just_exit.sh example. I noticed that I was running a different script actually.

By the way, the post_install script is run by `/bin/sh`, and it ignores the shebang, making some bash-specific syntax to cause errors. Also, python script does not run either. I use `base_os = ubuntu1404`.

So this line should be changed to something like  `(chmod +x $tmpfile ; exec $tmpfile $args )`
https://github.com/awslabs/cfncluster-cookbook/blob/master/files/default/fetch_and_run#L47

https://raw.githubusercontent.com/guicho271828/torque-utils/master/setup/test.py
```python
#!/usr/bin/env python

print "hi!"
```

local test (on my ubuntu 14.04 machine):
```
[guicho setup]$ ./test.py
hi!
[guicho setup]$ sh ./test.py
Warning: unknown mime-type for "hi!" -- using "application/octet-stream"
Error: no such file "hi!"
[guicho setup]$ ( exec ./test.py )
hi!
```

cfn-init.log: https://gist.github.com/guicho271828/67dc5cc6097dc0ea11d6cfc5a8204222
```
2016-05-27 04:04:52,658 [ERROR] Command runpostinstall (/opt/cfncluster/scripts/fetch_and_run -postinstall) failed
2016-05-27 04:04:52,658 [DEBUG] Command runpostinstall output: Warning: unknown mime-type for "hi!" -- using "application/octet-stream"
Error: no such file "hi!"
cfncluster: fetch_and_run - Failed to run boot_as_master postinstall
```

so the same symptom.

**Comment by dougalb** [comment]
We'll look at making those changes. Also happy to accept a pull request. It is the `fetch_and_run` file that executes the pre/post install scripts - https://github.com/awslabs/cfncluster-cookbook/blob/develop/files/default/fetch_and_run

**Comment by guicho271828** [comment]
yes, already done.

**Comment by dougalb** [comment]
This has been included in cfncluster-1.3.
